File: SDKs/probely.com/1.2.0/typescript/src/sdk/models/operations/getintegrationsjiracloudprojects.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetIntegrationsJiraCloudProjects401ApplicationJson extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=detail" })
  detail?: string;
}


export class GetIntegrationsJiraCloudProjects403ApplicationJson extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=detail" })
  detail?: string;
}


export class GetIntegrationsJiraCloudProjectsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata({ elemType: shared.JiraProject })
  jiraProjects?: shared.JiraProject[];

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  getIntegrationsJiraCloudProjects401ApplicationJSONObject?: GetIntegrationsJiraCloudProjects401ApplicationJson;

  @SpeakeasyMetadata()
  getIntegrationsJiraCloudProjects403ApplicationJSONObject?: GetIntegrationsJiraCloudProjects403ApplicationJson;
}

File: SDKs/probely.com/1.2.0/python/src/sdk/models/operations/get_integrations_jira_cloud_projects_.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import jiraproject as shared_jiraproject


@dataclass_json
@dataclasses.dataclass
class GetIntegrationsJiraCloudProjects401ApplicationJSON:
    detail: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('detail') }})
    

@dataclass_json
@dataclasses.dataclass
class GetIntegrationsJiraCloudProjects403ApplicationJSON:
    detail: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('detail') }})
    

@dataclasses.dataclass
class GetIntegrationsJiraCloudProjectsResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    jira_projects: Optional[list[shared_jiraproject.JiraProject]] = dataclasses.field(default=None)
    get_integrations_jira_cloud_projects_401_application_json_object: Optional[GetIntegrationsJiraCloudProjects401ApplicationJSON] = dataclasses.field(default=None)
    get_integrations_jira_cloud_projects_403_application_json_object: Optional[GetIntegrationsJiraCloudProjects403ApplicationJSON] = dataclasses.field(default=None)
    
